trestle/core/models/elements.py

```diff
@@ -58,31 +61,101 @@ def _parse(self, element_path: str) -> List[str]:
         """Parse the element path and validate."""
         parts: List[str] = element_path.split(self.PATH_SEPARATOR)
 
-        for i, part in enumerate(parts):
+        for part in parts:
             if part == '':
                 raise TrestleError(
                     f'Invalid path "{element_path}" because there are empty path parts between "{self.PATH_SEPARATOR}" '
                     'or in the beginning'
                 )
-            elif part == self.WILDCARD and i != len(parts) - 1:
-                raise TrestleError(f'Invalid path. Wildcard "{self.WILDCARD}" can only be at the end')
-
-        if parts[-1] == self.WILDCARD:
-            if len(parts) == 1:
-                raise TrestleError(f'Invalid path {element_path} with wildcard.')
-
-        if len(parts) <= 1:
-            raise TrestleError(
-                'Element path must have at least two parts with the first part being the model root name, '
-                'like "target-definition.metadata"'
-            )
 
+        if parts[0] == self.WILDCARD:
+            raise TrestleError(f'Invalid path {element_path} with wildcard.')
         return parts
 
     def get(self) -> List[str]:
         """Return the path parts as a list."""
         return self._path
 
+    def get_type(self, root_model: Optional[Type[OscalBaseModel]] = None, use_parent: bool = False) -> Type[Any]:
+        """Get the type of an element.
+
+        If possible the model type will be derived from one of the top level models,
+        otherwise a 'root model' can be passed for situations where this is not possible.
+
+        This type path should *NOT* have wild cards in it. It *may* have* indices.
```

Comment:
Please provide examples here so it is always clear what these paths look like.  Allowed and disallowed types.